tests/lib/tools/tests.session

```diff
@@ -121,9 +121,17 @@ case "$action" in
 				;;
 		esac
 
-		# Work around https://github.com/systemd/systemd/issues/12401 fixed
-		# in systemd v243.
-		if systemctl cat systemd-logind.service | not grep -q StateDirectory; then
+		# Try to enable linger for the selected user(s).
+		needs_linger_workaround=
+		for u in $(echo "$user" | tr ',' ' '); do
+			if ! loginctl enable-linger "$u"; then
+				needs_linger_workaround=true
+			fi
+		done
+
+		# Work around https://github.com/systemd/systemd/issues/12401 fixed in
+		# systemd v243.
+		if [ "$needs_linger_workaround" = "true" ] && systemctl cat systemd-logind.service | not grep -q StateDirectory; then
```

Comment:
So once you know you need the linger workaround where is the code that re-tries the linger setting with the workaround in place?